src/backend/joanie/core/models/wishlist.py

```diff
@@ -0,0 +1,37 @@
+"""
+Declare and configure the models for the wishlist part
+"""
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from . import Course, User
+from .base import BaseModel
+
+
+class CourseWish(BaseModel):
+    """
+    CourseWish represents and records a user wish to participate at a course
+    """
+
+    owner = models.ForeignKey(
+        to=User,
+        verbose_name=_("Owner"),
+        related_name="wishlists",
+        on_delete=models.PROTECT,
+    )
+
+    course = models.ForeignKey(
+        to=Course,
+        verbose_name=_("Course"),
+        related_name="wished_in_wishlists",
```

Comment:
I am not sure about `wishlist` for `related_name` because, it can return wishes from different users' wishlists.
It's return a list of all wishes without filtering by user where the `course` is linked.